lib/reverse_sentence.rb

```diff
@@ -1,6 +1,26 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+
+  if my_sentence == nil
+    return ""
+  end
+  # having trouble figuring out why the trailing whitespace isn't being preserved, but the rest is?
+  words = my_sentence.split(/ /)
+
+  i = 0
+  j = words.length - 1
+  while i < j
+    temp = words[i]
+    words[i] = words[j]
+    words[j] = temp
+    i += 1
+    j -= 1
+  end
+
+  word_str = words.join(" ")
```

Comment:
This part here assumes that the whitespace is all one blank space.  Could you use regex to split by whitespace in general and include the whitespace in the splitted string?